test/src/test/java/jenkins/health/HealthCheckActionTest.java

```diff
@@ -0,0 +1,199 @@
+                ]
+            }
+            """), JSONObject.fromObject(page.getWebResponse().getContentAsString()));
+        }
+    }
+
+    @TestExtension({"healthCheckSuccessExtension", "healthCheckFailingExtension"})
+    public static class SuccessHealthCheck implements HealthCheck {
+
+        @Override
+        public String getName() {
+            return "success";
+        }
+
+        @Override
+        public boolean check() {
+            return true;
+        }
+    }
+
+    @Test
+    public void healthCheckFailingExtension() throws Exception {
+        try (var webClient = r.createWebClient()) {
+            webClient.getOptions().setThrowExceptionOnFailingStatusCode(false);
+            webClient.getOptions().setPrintContentOnFailingStatusCode(false);
+            var page = webClient.goTo("healthCheck", "application/json");
+            assertThat(page.getWebResponse().getStatusCode(), is(503));
+            assertEquals(JSONObject.fromObject("""
+            {
+                "status": false,
+                "data": [
+                  {
+                    "name": "completedInitialization",
+                    "result": true
+                  },
+                  {
+                    "name": "failing",
+                    "result": false
+                  },
+                  {
+                    "name": "success",
+                    "result": true
+                  }
+                ]
+            }
+            """), JSONObject.fromObject(page.getWebResponse().getContentAsString()));
+        }
+    }
+
+    @TestExtension("healthCheckFailingExtension")
+    public static class FailingHealthCheck implements HealthCheck {
+
+        @Override
+        public String getName() {
+            return "failing";
+        }
+
+        @Override
+        public boolean check() {
+            return false;
+        }
+    }
+
+    @Test
+    public void duplicateHealthCheckExtension() throws Exception {
+        try (var webClient = r.createWebClient()) {
+            var page = webClient.goTo("healthCheck", "application/json");
+            assertThat(page.getWebResponse().getStatusCode(), is(200));
+            assertEquals(JSONObject.fromObject("""
+            {
+                "status": true,
+                "data": [
+                  {
+                    "name": "completedInitialization",
+                    "result": true
+                  },
+                  {
+                    "name": "dupe",
+                    "result": true
+                  }
+                ]
+            }
+            """), JSONObject.fromObject(page.getWebResponse().getContentAsString()));
+        }
+        // TestExtension doesn't have ordinal, but I think by default extensions are ordered alphabetically
+        assertThat(loggingRule.getMessages(), contains("Ignoring duplicate health check with name dupe from " + HealthCheck2.class.getName() + " as it is already defined by " + HealthCheck1.class.getName()));
```

Comment:
IMO the duplicate-detection code is overkill; the Javadoc warning should suffice.